tensorboard/webapp/persistent_settings/_data_source/types.ts

```diff
@@ -0,0 +1,35 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+/**
+ * Global settings that the backend remembers. `declare`d so it does not get
+ * mangled or mangled differently when compiler changes.
+ */
```

Comment:
the "compiler" here means it is either localstorage or (future backend support?)

what are the "keys" in your mind that these two interfaces could be different?
(then we need a converter for it)

Reply:
Clarified the "compiler" bit. PTAL.